### ISSUE TITLE
My Gated Builds are failing because uploading the test results appear to time out

### ISSUE THREAD

**Issue by M-OpsJoe** [Microsoft/azure-pipelines-tasks#8690]
## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

## Environment
- Server - Azure Pipelines 
    
    - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: 

- Agent - Private: 
    
    - If using private agent, provide the OS of the machine running the agent and the agent version: Windows 2012 R2 all agents on same machine
[log_10_81941.zip](https://github.com/Microsoft/azure-pipelines-tasks/files/2508974/log_10_81941.zip)
[log_10_81941.zip](https://github.com/Microsoft/azure-pipelines-tasks/files/2508982/log_10_81941.zip)

First Agent
Agent.OSVersion | 6.3.9600
Agent.Version | 2.126.0

Second and Third Agents
Agent.OSVersion | 6.3.9600
Agent.Version | 2.140.2

## Issue Description

Unable to process command '##vso[telemetry.publish area=TestExecution;feature=TestExecutionTask;]{"builduri":"vstfs:///Build/Build/81939","buildid":"81939","areacode":"ExecuteVsTest","result":"C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\Common7\\IDE\\CommonExtensions\\Microsoft\\TestWindow\\vstest.console.exe failed with return code: 1","tracepoint":1005,"isusererror":true}' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)

An error occurred while sending the request.

Error: C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe failed with return code: 1

VsTest task failed.

### Task logs

[Enable debug logging and please provide the zip file containing all the logs for a speedy resolution]

### Error logs

[Insert error from the logs here for a quick overview]

**Comment by kaadhina** [comment]
The build failed because vstest.console has reported the test run to have failed:
2018-10-24T04:26:49.0749362Z Test Run Failed.

I can see some error logs from the console output that most likely caused this:
2018-10-24T04:26:31.4383528Z Error: Error: Error opening C:/agent/_work/6/b/_PublishedWebsites/MarineEnterprise.Web/undefinedapi/FlightScheduler/CancelScheduledEvent: The system cannot find the path specified.
2018-10-24T04:26:31.4384049Z While Running:C:\agent\_work\6\b\_PublishedWebsites\MarineEnterprise.Web\specs\flight\shared\scheduledEventCancel.spec.js
You can also take a look at the trx (from test run attachment) to see if that indicated what went wrong with the run. Fixing these errors in the test run should make the task pass.

As for the error mentioned above, it is the telemetry publish that seems to have failed (which was then ignored and the task continued). The test results publish succeeded:
2018-10-24T04:27:06.7644987Z Published Test Run

**Comment by M-OpsJoe** [comment]
Thank you very much. Those errors have been resolved but the build is still failing with a few warnings but the errors listed below.

Unable to process command '##vso[results.publish type=VSTest;mergeResults=false;platform=x64;config=release;runTitle=AzureDevOps_SeaMonkey_Gated_20181025.1 Integration Tests;publishRunAttachments=false;resultFiles=C:\agent\_work\6\s\TestResults\BuildAgent.svc_MSE-BLD-AGT01 2018-10-25 07_39_52.trx;]' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)

An error occurred while sending the request.

**Comment by kaadhina** [comment]
Can you please attach the full debug logs for this one?

**Comment by M-OpsJoe** [comment]
Hello, thanks for your help.  The logs are attached.

**Comment by kaadhina** [comment]
Thank you.
Is a consistent failure? From the two logs, I see that different calls failed at different places with error:
2018-10-24T04:27:01.1230334Z ##[debug]System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.Http.WinHttpException: The server name or address could not be resolved
Is there a network issue in the agent machines?

**Comment by M-OpsJoe** [comment]
Thanks for all the help @Kaadhina.  We confirmed there are no issues with the network connectivity though we did come back to it several times.  Currently it looks like that particular agent might just have gone bad? We have 3 agents on the same machine (this one is the oldest, though not by much) and I disabled it last night and we have had several successful builds now.  I'm waiting until a few more complete to actually point the finger solely at that agent.  It's just strange that those type of "unable to connect" issues would come from the same machine with the same configurations of the agents.
